docs/OpenSSLSupport.md

```diff
@@ -148,6 +148,31 @@ Note that the code snippet for the RDRAND engine opt-in is required to use not o
+
+OpenSSL provides APIs that allow users to configure sensitive settings like certificate trust and cipher suite preference from files.
+Because the host file system is considered untrusted in contexts such as SGX enclaves, OE SDK marks these APIs as unsupported to discourage their use.
+OE SDK does this by patching the OpenSSL headers that include such APIs or macros (i.e., appending the inclusion of an "unsupported.h" file to these headers).
+This ensures that when an enclave includes a patched OpenSSL header and uses the specific API or macro, user will receive compile-time errors.
+The errors can be disabled by specifying the `OE_OPENSSL_SUPPRESS_UNSUPPORTED` option to the compiler.
+
+See the following table for the detailed list of APIs/Macros.
+
+API / Macro | Original header | Comments | Guidance |
+:---:|:---:|:---|:---|
+OPENSSL_INIT_LOAD_CONFIG | crypto.h | The macro represents an option to the OPENSSL_init_ssl and OPENSSL_init_crypto APIs that initializes an application based on the openssl.cnf (loaded from the host filesystem). Therefore, the use of this option would allow an untrusted host to fully control the initialization of the application. | The recommendation is not to invoke initialization APIs with this option. Note that starting from v1.1.0, the explicit initialization is not required. |
+SSL_CTX_load_verify_locations | ssl.h | The API specifies the default locations from which an application looks up CA certificates for verification purposes. The API would allow an untrusted host to fully control what certificates the application will trust. | The recommendation is using the SSL_CTX_set_cert_store API that specifies an in-memory certificate verification storage (`X509_STORE`). Another alternative is to implement a customized verification callback and sets the callback via the SSL_CTX_set_verify API, which effectively bypasses the default implementation. |
+SSL_CTX_set_default_verify_paths | ssl.h | The API specifies the locations as part of arguments from which an application looks up CA certificates for verification purposes. The API would allow an untrusted host to fully control what certificates the application will trust. | Same recommendation as SSL_CTX_load_verify_locations. |
+SSL_CTX_set_default_verify_dir | ssl.h | Similar to SSL_CTX_set_default_verify_paths except for specifying dir only. | Same recommendation as SSL_CTX_load_verify_locations. |
+SSL_CTX_set_default_verify_file | ssl.h | Similar to SSL_CTX_set_default_verify_paths except for specifying file only. | Same recommendation as SSL_CTX_load_verify_locations. |
+X509_LOOKUP_hash_dir | x509_vfy.h | This API returns a `X509_LOOKUP_METHOD` method that loads files from the path specified by the `SSL_CERT_DIR` environment variable. The API is used by SSL_CTX_set_default_verify_dir and SSL_CTX_set_default_verify_paths internally. | The recommendation is implementing a customized `X509_LOOKUP_METHOD` method based on in-memory operations. |
```

Comment:
>> Might be useful to clarify that the X509_LOOKUP methods are not blocked because they are used SSL_CTX_set_default_verify_* methods; they're blocked because they provide analogous functionality to X509_load_*_file methods.
>
>I added a recommendation to implement a customized X509_LOOPUP_METHOD. Will that be sufficient?

My previous comment here was with regards to the Comment column, not the newly added Guidance column; i.e. the Comments should explain why the API is blocked (it provides way to load trust certificates and CRLs from untrusted host) rather than functional details (it is used in the SSL_CTX_* methods). 

Regarding the guidance, I think that's fine, though I would suggest adding a bit more context to the recommendation. For example:
```md
This method is used internally by `SSL_CTX_set_default_verify_dir` and `SSL_CTX_set_default_verify_paths`.
You can define your own enclave-specific [`X509_LOOKUP_METHOD`](
https://www.openssl.org/docs/man1.1.1/man3/X509_LOOKUP_meth_new.html)
but it will not be used by the SSL_CTX methods transparently. We recommend
explicitly setting certificate trust store witch `SSL_CTX_set_cert_store` instead.
```

Reply:
Updated.

Reply:
Note: the cross reference does not seem to work in the table.